source/MaterialXContrib/MaterialXNode/MaterialXUtil.cpp

```diff
@@ -50,5 +50,47 @@ mx::DocumentPtr loadDocument(const std::string& materialXDocumentPath,
     return document;
 }
 
+std::vector<mx::TypedElementPtr> getRenderableElements(mx::DocumentPtr document, const std::string &desiredElementPath)
```

Comment:
New renderable element determination which can be migrated to core afterwards like the rest of these methods. The "find" in renderable list should be part of core logic.

Reply:
TBH I find this API extremely confusing. At the very least, the name is a misnomer - it doesn't just return renderable elemtents. There are so many potential outcomes as well. E.g. if there are no renderable elements - why not just return an empty vector?

Reply:
In general, there two distinct cases here: we either want all renderable elements OR we want to return a single element with the specific name if it's renderable. I'd much rather have two separate functions: one that returns all renderables as a vector, which would be called in the case when we want to instantiate all renderables in the document and another one calling it that would just check if the element is renderable and return a single element not bothering with vectors. This way, the two functions can have perfectly descriptive names (getRenderableElements and getRenderableElement?). The first one wouldn't need the desiredElementPath argument at all, so there would be no need to document or guess that an empty path actually means "return all renderables".